homeassistant/components/tradfri/cover.py

```diff
@@ -22,33 +23,38 @@ async def async_setup_entry(
 ) -> None:
     """Load Tradfri covers based on a config entry."""
     gateway_id = config_entry.data[CONF_GATEWAY_ID]
-    tradfri_data = hass.data[DOMAIN][config_entry.entry_id]
-    api = tradfri_data[KEY_API]
-    devices = tradfri_data[DEVICES]
+    coordinator_data = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
+    api = coordinator_data[KEY_API]
 
     async_add_entities(
-        TradfriCover(dev, api, gateway_id) for dev in devices if dev.has_blind_control
+        TradfriCover(
+            device_coordinator,
+            api,
+            gateway_id,
+        )
+        for device_coordinator in coordinator_data[COORDINATOR_LIST]
+        if device_coordinator.device.has_blind_control
     )
 
 
-class TradfriCover(TradfriBaseDevice, CoverEntity):
+class TradfriCover(TradfriBaseEntity, CoverEntity):
     """The platform class required by Home Assistant."""
 
     def __init__(
         self,
-        device: Command,
+        device_coordinator: TradfriDeviceDataUpdateCoordinator,
         api: Callable[[Command | list[Command]], Any],
         gateway_id: str,
     ) -> None:
-        """Initialize a cover."""
-        self._attr_unique_id = f"{gateway_id}-{device.id}"
-        super().__init__(device, api, gateway_id)
-        self._refresh(device, write_ha=False)
+        """Initialize a switch."""
+        super().__init__(
+            device_coordinator=device_coordinator,
+            api=api,
+            gateway_id=gateway_id,
+        )
 
-    @property
-    def extra_state_attributes(self) -> dict[str, str] | None:
```

Comment:
Do we really need this? Isn't it handled by the base entity?

Reply:
It was added here to be able to use the model in automations.

https://github.com/home-assistant/core/pull/33674

I don't think we should remove it in this PR. We can look at it separately.

Reply:
OK, I've put them back.